lib/Net/Async/Blockchain/ETH.pm

```diff
@@ -599,9 +597,63 @@ sub get_numeric_from_hex {
 
     # numeric responses should have at least 64 + 2(0x) = 66 characters
     # transaction data field / contract response
-    return undef unless ($hex && length($hex) == 66);
+    return undef
+        unless ($hex && $hex =~ /^0x[0-9a-fA-F]{64}$/);
 
     return Math::BigFloat->from_hex($hex);
 }
 
+=head2 get_valid_decimal_places
+
+Check the given decimal places contract response following the steps:
+- hex is valid
+- hex is a numeric value
+- is a valid Math::BigFloat not 0
+- is between the limit of 18 decimal places
+
+=over 4
+
+=item * C<decimal_places> contract decimal places
+
+=back
+
+If the hex is valid returns a L<Math::BigFloat> object
+
+=cut
+
+sub get_valid_decimal_places {
+    my ($self, $decimal_places) = @_;
+
+    my $float = $self->get_numeric_from_hex($decimal_places);
+    # decimal places can be 0 and should follow the limit of 18 decimal places
+    return undef unless eval { $float->isa('Math::BigFloat') } && $float->ble(DEFAULT_DECIMAL_PLACES);
```

Comment:
Are we sure `->ble(DEFAULT_DECIMAL_PLACES)` is what we want here?

Reply:
Yes, the limit for decimal places so if this condition is true I don't want to return undef